File: pingyougoumanagerweb/src/main/webapp/js/controller/BrandController.js
app.controller('brandController', function ($scope, brandService,$controller) {

    $controller('baseController',{$scope:$scope});

    $scope.queryAll = function () {
        brandService.queryAll('../brand/queryAll.do').success(
            function (response) {
                $scope.list = response;
            }
        );
    }
    // 分页控件配置

    $scope.fillPage = function (page, size) {
        brandService.fillPage(page, size).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }
        );
    }

    $scope.save = function () {
        var obj = null;
        if ($scope.entity.id != null) {
            obj = brandService.update($scope.entity);
        } else {
            obj = brandService.add($scope.entity);
        }
        obj.success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();
                } else {
                    alert(response.message);
                }
            }
        );
    }

    $scope.queryOne = function (id) {
        brandService.queryOne(id).success(
            function (reponse) {
                $scope.entity = reponse;
            }
        );
    }

    $scope.remove = function () {
        brandService.remove($scope.ids).success(
            function (reponse) {
                if (reponse.success) {
                    $scope.reloadList();
                } else {
                    alert(reponse.message);
                }
            }
        );
    }

    $scope.search = function (page, size) {
        brandService.search(page, size, $scope.domain).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }
        );
    }
    

});

File: pingyougoumanagerweb/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageBean;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {


    @Reference
    private BrandService brandService;

    @RequestMapping("/queryAll")
    public List<TbBrand> queryAll(){
        return brandService.queryAll();
    }


    @RequestMapping("/fillPage")
    public PageBean fillPage(int page,int size) {
        return brandService.fillPage(page, size);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return new Result(true,"新增成功");
        } catch (Exception e) {
            return new Result(false,"新增失败");
        }
    }

    @RequestMapping("/queryOne")
    public TbBrand queryOne(long id){
       return brandService.queryOne(id);
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand) {
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("/delete")
    public Result delete(long[] ids) {
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("/search")
    public PageBean search(@RequestBody TbBrand brand,int page,int size) {
        return brandService.fillPage(brand,page,size);
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }

}
